Mehdi_test/date_f.nit

```diff
@@ -0,0 +1,104 @@
+#Module date_f has the basic date fuction implementation
+module date_f
+
+#The main class Date
+public class Date
+
+private var years_In : Int 
+private var mounth_In : Int
+private var day_In : Int
+
+private var hours_In : nullable Int = 0
+private var min_In : nullable Int = 0
+private var sec_In : nullable Int = 0
+
+	#Return the with String type
+	redef fun to_s do
+
+	var y = years_In.to_s
+	var m = mounth_In.to_s
+	var d = day_In.to_s
+
+	return y +"/"+ m +"/"+ d
+	end
+	
+	#Compare two Date
+	# forms : 1 for day, 2 for mounth, 3 for years
+	fun date_diff( d2 : Date, forms : Int ) : Int do
+
+		var y_Out  = 0
+		var m_Out  = 0
+		#var d_Out  = 0
+
+		var out = 0
+
+		if forms == 1 then 
+			y_Out = years_In - d2.get_y
+			y_Out = y_Out * 365
+
+			m_Out = mounth_In - d2.get_m
+			m_Out = m_Out * 30
+
+			out = day_In - d2.get_d + m_Out + y_Out
+			
+	
+		else if forms == 2 then
+			y_Out = years_In - d2.get_y
+			y_Out = y_Out * 12
+
+			out = mounth_In - d2.get_m + y_Out
+		end
+		
+		return out
+
+	end
+
+	#equals: this function compare two date equality
```

Comment:
Generally, you can avoid saying "This function", "This attribute" etc.

For example you can just say: "Is `d` equals to `self`?"